### PR TITLE
feat: Add <LoggedInAs /> component

### PR DESCRIPTION
**Pull request by joshlarson**
Screenshot 2024-02-09 at 8 37 48 PM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206526459943484

### CODE REVIEW

File: assets/src/components/loggedInAs.tsx
Comment (joshlarson) [inline assets/src/components/loggedInAs.tsx:5]: This hacks in the username icon; whenever that's ready, we should use that here instead.

File: assets/css/_logged_in_as.scss
Comment (lemald) [inline assets/css/_logged_in_as.scss:8]: **question:** if I'm not mistaken this is the new eggplant-500 that we're not actually using yet, so I guess there's a question of which to use for now. I'm asking design in the context of my own ticket FWIW.

Comment (lemald) [reply assets/css/_logged_in_as.scss:8]: Oh heh true.

File: assets/src/components/loggedInAs.tsx
Comment (joshlarson) [reply assets/src/components/loggedInAs.tsx]: No more! https://github.com/mbta/skate/pull/2383/commits/8fb447e8adc48f1158038538c65f40796cb520eb

File: assets/css/_logged_in_as.scss
Comment (joshlarson) [reply assets/css/_logged_in_as.scss]: 👋 https://github.com/mbta/skate/pull/2383/commits/8fb447e8adc48f1158038538c65f40796cb520eb

File: assets/css/_logged_in_as.scss
Comment (joshlarson) [inline assets/css/_logged_in_as.scss:15]: As @lemald pointed out about a different color (that got `git`-ed away), this color may want to be replaced with `$color-gray-700`, although if we do that before [this PR](https://github.com/mbta/skate/pull/2365) is merged, then the color will be slightly wrong, since `$color-gray-700` is currently #3c3f4c

Comment (joshlarson) [reply assets/css/_logged_in_as.scss:15]:
```suggestion
  color: $color-gray-700;
```

File: assets/css/_logged_in_as.scss
Comment (lemald) [inline assets/css/_logged_in_as.scss:14]: **suggestion:** rather than hardcoding these you can use values defined in `_skate_ui.scss`.

File: assets/css/_logged_in_as.scss
Comment (lemald) [inline assets/css/_logged_in_as.scss:9]: **suggestion:** (here and a few other places) we usually use `rem` for sizing, except within Leaflet maps because those already use `px` for a lot of the sizing.

**Comment by lemald** [comment]
I forgot to add yesterday - I'm not approving yet because of some of my comments on the CSS, but aside from that I've looked over the code and it all looks good to me. 👍

File: assets/css/_logged_in_as.scss
Comment (firestack) [inline assets/css/_logged_in_as.scss:18]: question: can unitless values be used instead here?
[developer.mozilla.org/en-US/docs/Web/CSS/line-height#prefer_unitless_numbers_for_line-height_values](https://developer.mozilla.org/en-US/docs/Web/CSS/line-height#prefer_unitless_numbers_for_line-height_values)

File: assets/css/_logged_in_as.scss
Comment (joshlarson) [reply assets/css/_logged_in_as.scss:15]: On further thought - the color is more than slightly wrong - the other links on this page are `gray-600`, not `gray-700`, so the text winds up looking pretty out of place. I think we should keep this as-is and follow up after the bootstrap colors PR is merged.

File: assets/css/_logged_in_as.scss
Comment (joshlarson) [reply assets/css/_logged_in_as.scss:9]: Ahh good. I was wondering about that, and planning to ask.

File: assets/css/_logged_in_as.scss
Comment (joshlarson) [reply assets/css/_logged_in_as.scss:18]: Screenshot 2024-02-13 at 10 51 43 AM

Nailed it

File: assets/css/_logged_in_as.scss
Comment (joshlarson) [reply assets/css/_logged_in_as.scss]: Ahhh cool - https://github.com/mbta/skate/pull/2383/commits/e40fec99962d5201d3358ef704403ebc094acb1c

File: assets/css/_logged_in_as.scss
Comment (joshlarson) [reply assets/css/_logged_in_as.scss]: Okay but here we go... doing it correctly! https://github.com/mbta/skate/pull/2383/commits/e40fec99962d5201d3358ef704403ebc094acb1c

**Review by firestack — APPROVED** [review]

File: assets/css/_logged_in_as.scss
Comment (lemald) [reply assets/css/_logged_in_as.scss:15]: Per Slack discussion: let's go with gray-700 for this PR since we're doing other work to integrate the new color palette changes.

Comment (joshlarson) [reply assets/css/_logged_in_as.scss:15]: If I'm synthesizing correctly from all of the other discussions, I'm going to wait for https://github.com/mbta/skate/pull/2365 to get merged, and then use the new_v2 (or 2024) gray-700.

Comment (joshlarson) [reply assets/css/_logged_in_as.scss]: Aaaand here we go: https://github.com/mbta/skate/pull/2383/commits/9c5ea5b2c3bc03fbf1381c07477966eb00e5421c

File: assets/css/_logged_in_as.scss
Comment (lemald) [inline assets/css/_logged_in_as.scss:6]: **nitpick:** this is probably less of an issue two years on, but we have run into problems with `gap` in flexbox in the past. See #1541.

**Review by lemald — APPROVED** [review]
You may want to change how you accomplish the spacing in the flexbox to avoid the use of `gap` per my above comment, but it's been a couple of years since that was last an issue so I'll leave it up to your discretion.

File: assets/css/_logged_in_as.scss
Comment (firestack) [reply assets/css/_logged_in_as.scss:6]: We really should get caniuse in CI (again?)

Comment (joshlarson) [reply assets/css/_logged_in_as.scss:6]: Oof! Browser incompatibility strikes again!

Would you recommend switching to something like `margin-right: 0.5rem` on the icon?

Comment (joshlarson) [reply assets/css/_logged_in_as.scss:6]: Or (I think equivalently) `margin-left: 0.5rem` on the text?

Comment (joshlarson) [reply assets/css/_logged_in_as.scss]: https://github.com/mbta/skate/pull/2383/commits/04458340325ba72ec0daefe42ae7a89574684d13

**Pull request merged**
